app/gui/src/controller/searcher.rs

```diff
@@ -720,12 +730,64 @@ impl Searcher {
         self.notifier.notify(Notification::NewActionList);
     }
 
+    const AI_QUERY_PREFIX: &'static str = "AI:";
+    const AI_QUERY_ACCEPT_TOKEN: &'static str = "#";
+    const AI_STOP_SEQUENCE: &'static str = "`";
+    const AI_GOAL_PLACEHOLDER: &'static str = "__$$GOAL$$__";
+
+    async fn accept_ai_query(
```

Comment:
I know this is PoC, but docs are needed here for future refactoring/development. Please describe shortly what this function does and why.